### PR TITLE
Add a new target HolybroF4_PMU

### PR DESCRIPTION
**Pull request by jamming**
Add a new target HolybroF4_PMU. It uses STM32F405 MCU and ALLEGRO ACS772 current sensor.

### CODE REVIEW

**Comment by vincentpoont2** [comment]
Hi @peterbarker @tridge , This is for our Holybro DroneCAN Power Module. 
we have tested the preset parameter is working with this https://github.com/ArduPilot/ardupilot/pull/27735.

Please review and merge.

File: libraries/AP_HAL_ChibiOS/hwdef/HolybroF4_PMU/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+- Power & CAN Port: Molex CLIK-Mate 2mm 6Pin
+- Battery IN/OUT Options:
+     --XT90 Connectors
+     --Tinned Wires
+     --XT90 & Ring Terminals
+
+## Status LED
+
+- Flashing quickly continuously: MCU is in the bootloader stage, waiting for firmware to be flash
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/HolybroF4_PMU/README.md:31]:
```suggestion
- Flashing quickly continuously: MCU is in the bootloader stage, waiting for firmware to be flashed

```

File: libraries/AP_HAL_ChibiOS/hwdef/HolybroF4_PMU/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+- Flashing quickly for 3 seconds, and then on for 1 second: Waiting for CAN connection
+- Flashing slowly (one-second intervals): CAN is successfully connected
+
+## Mechanical Spec
+
+- Size: 45mm��41mm��26mm (Not Include Cable)
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/HolybroF4_PMU/README.md:37]:
```suggestion
- Size: 45mm��41mm��26mm (not including cable)

```

File: libraries/AP_HAL_ChibiOS/hwdef/HolybroF4_PMU/hwdef.dat [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+# crystal frequency
+OSCILLATOR_HZ 16000000
+
+STDOUT_SERIAL SD3
+STDOUT_BAUDRATE 57600
+
+# blue LED0 marked as ACT
+PB0 LED OUTPUT HIGH
+define HAL_LED_ON 0
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/HolybroF4_PMU/hwdef.dat:32]: I don't think this LED will blink at the moment.  You need to set things like this now:
define AP_NOTIFY_GPIO_LED_1_ENABLED 1
define AP_NOTIFY_GPIO_LED_1_PIN 90

File: libraries/AP_HAL_ChibiOS/hwdef/HolybroF4_PMU/hwdef.dat [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+# avoid timer and RCIN threads to save memory
+define HAL_NO_RCIN_THREAD
+
+define HAL_USE_RTC FALSE
+define DISABLE_SERIAL_ESC_COMM TRUE
+
+define DMA_RESERVE_SIZE 0
+
+define HAL_DISABLE_LOOP_DELAY
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/HolybroF4_PMU/hwdef.dat:50]: No longer used - this line is not effective

File: libraries/AP_HAL_ChibiOS/hwdef/HolybroF4_PMU/hwdef.dat [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+define HAL_NO_MONITOR_THREAD
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/HolybroF4_PMU/hwdef.dat:52]: This line doesn't do anything

**Review by peterbarker — CHANGES_REQUESTED** [review]
There should be two commits here.

One for the bootloaders directory, e.g.: `bootloaders: add HolybroF4_PMU bootloader`
One for the hwdef directory: `hwdef: add HolyBroF4_PMU`

@Hwurzburg is currently on holidays but does like to review these PRs before they're merged.

**Comment by jamming** [comment]
I have already made all the modifications according to your suggestions.

**Review by Hwurzburg — APPROVED** [review]

**Comment by vincentpoont2** [comment]
Hi @peterbarker is it good to merge now?

**Comment by peterbarker** [comment]
I've created https://github.com/ArduPilot/ardupilot/pull/27998 to replace this.

I was unable to force-push over this branch - probably a repo setting or someone unchecked the "allow force push" option when creating the PR.

The commits needed fixing to conform to our standards - check that other PR to see the shape they need to be.

Please note that the many-repo pattern you're using for these developments is a little vexing to work with as the remote needs to be updated for this sort of tidy-up.

I've marked my other PR as merge-on-ci-pass, so I'm going to close this one.

Thanks!

**Comment by peterbarker** [comment]
> Hi @peterbarker is it good to merge now?

I've merged the PR I created to replace this one.

**Pull request closed without merging**